test/sync/src/commonMain/kotlin/io/realm/test/mongodb/TestApp.kt

```diff
@@ -85,6 +87,15 @@ class TestApp(
         deleteAllUsers()
     }
 
+    suspend fun createUserAndLogIn(email: String, password: String): User =
```

Comment:
I think it would be nice if we could hold all methods using our actual API on the `App` and leave the `TestApp` for stuff that is only related to test. At least that was the motivation of making the explicit `app.asTestApp()` entry point, as I found it confusing in `realm-java` that we had a lot of extension methods on `App` with test infrastructure code. 

All these methods are practically just using our actual API, so could just be extension methods on `App`.